### PR TITLE
Add option not to fail if the credentials can't be found

### PR DESCRIPTION
**Pull request by MichaelSp**
Reason for this:
If we have convention over configuration and the convention allows several places where the credentials can be stored, we need to gracefully go through the list of possible locations. Failing is still possible if you check if the resulting env is empty

Signed-off-by: Michael Sprauer <Michael.Sprauer@sap.com>

### CODE REVIEW

**Comment by ptierno** [comment]
@MichaelSp can you please rebase against the latest commits before i full review this?

**Comment by MichaelSp** [comment]
ping @ptierno re-based... pls review

File: src/main/java/com/datapipe/jenkins/vault/VaultAccessor.java [hunk 1]
```diff
@@ -17,8 +19,13 @@
...
 
     public void init(String url) {
+        init(url, false);
+    }
+
+    public void init(String url, boolean skipSslVerification) {
         try {
-            config = new VaultConfig().address(url).build();
+            config = new VaultConfig(url).sslVerify(skipSslVerification)
+                    .build();
```
Comment (ptierno) [inline src/main/java/com/datapipe/jenkins/vault/VaultAccessor.java]: Looking at the vault-java-driver code for 3.1.0 (currently in use) This should look like

```
config = new VaultConfig().address(url).sslConfig(new SslConfig().verify(skipSslVerification).build()).build()
```

**Comment by MichaelSp** [comment]
done

**Comment by ptierno** [comment]
@MichaelSp One more request. please rebasae off of master and I will merge.

**Comment by MichaelSp** [comment]
friendly hint:

**Comment by ptierno** [comment]
@MichaelSp 

```
This branch cannot be rebased due to conflicts
Rebasing the commits of this branch on top of the base branch cannot be performed automatically due to conflicts encountered while reapplying the individual commits from the head branch.
```

**Comment by MichaelSp** [comment]
ping @ptierno

**Comment by ptierno** [comment]
@MichaelSp Cant 'Rebase and merge' due to conflicts.

File: src/main/java/com/datapipe/jenkins/vault/VaultBuildWrapper.java [hunk 2]
```diff
@@ -127,24 +112,33 @@ public void setVaultAccessor(VaultAccessor vaultAccessor) {
...
+        vaultAccessor.init(url, credential, configuration.isSkipSslVerification());
         ArrayList<LogicalResponse> responses = new ArrayList<>();
         for (VaultSecret vaultSecret : vaultSecrets) {
-            LogicalResponse response = vaultAccessor.read(vaultSecret.getPath());
-            responses.add(response);
-            Map<String, String> values = response.getData();
-            for (VaultSecretValue value : vaultSecret.getSecretValues()) {
-                valuesToMask.add(values.get(value.getVaultKey()));
-                context.env(value.getEnvVar(), values.get(value.getVaultKey()));
+            vaultAccessor.auth(credential);
```
Comment (ptierno) [inline src/main/java/com/datapipe/jenkins/vault/VaultBuildWrapper.java]:
```
[ERROR] /Users/ptierno/Projects/java/hashicorp-vault-plugin/src/main/java/com/datapipe/jenkins/vault/VaultBuildWrapper.java:[127,26] cannot find symbol
  symbol:   method auth(com.datapipe.jenkins.vault.credentials.VaultCredential)
  location: variable vaultAccessor of type com.datapipe.jenkins.vault.VaultAccessor
```

The credential performs the auth, not the accessor.

https://github.com/jenkinsci/hashicorp-vault-plugin/blob/master/src/main/java/com/datapipe/jenkins/vault/credentials/VaultCredential.java#L15

Comment (MichaelSp) [reply src/main/java/com/datapipe/jenkins/vault/VaultBuildWrapper.java]: confused 😕 That line comes from here https://github.com/jenkinsci/hashicorp-vault-plugin/blob/3dda0cabd05c3cb8cad45ce85f7ab7b16a69aa86/src/main/java/com/datapipe/jenkins/vault/VaultBuildWrapper.java#L131
I didn't mean to change the logic...
I try to fix that again...

**Comment by MichaelSp** [comment]
not so easy to fix the tests, when they are not running locally ... but finally...

**Pull request merged**